src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -112,9 +118,220 @@ def export_helm_chart(registry_path, chart_export_path, kube_config, kube_contex
+                shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+            else:
+                logger.warning("An exception has occured while performing the DNS check on the cluster. Exception: {}".format(str(e)) + "\n")
+                telemetry.set_exception(exception=e, fault_type=consts.Cluster_DNS_Check_Fault_Type, summary="Error occured while performing cluster DNS check")
+                diagnoser_output.append("An exception has occured while performing the DNS check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+        # To handle any exception that may occur during the execution
+        except Exception as e:
+            logger.warning("An exception has occured while performing the DNS check on the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Cluster_DNS_Check_Fault_Type, summary="Error occured while performing cluster DNS check")
+            diagnoser_output.append("An exception has occured while performing the DNS check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+        return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+
+def check_cluster_outbound_connectivity(outbound_connectivity_check_log, for_preonboarding_checks=False, filepath_with_timestamp=None, storage_space_available=False):
+    if for_preonboarding_checks:
+        try:
+            outbound_connectivity_response = outbound_connectivity_check_log[-1:-4:-1]
+            outbound_connectivity_response = outbound_connectivity_response[::-1]
+            if consts.Outbound_Connectivity_Check_Result_String not in outbound_connectivity_check_log:
+                return consts.Diagnostic_Check_Incomplete
+            # Validating if outbound connectiivty is working or not and displaying proper result
+            if(outbound_connectivity_response != "000"):
+                return consts.Diagnostic_Check_Passed
+            else:
+                logger.warning("Error: We found an issue with outbound network connectivity from the cluster.\nIf your cluster is behind an outbound proxy server, please ensure that you have passed proxy parameters during the onboarding of your cluster.\nFor more details visit 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli#connect-using-an-outbound-proxy-server'.\nPlease ensure to meet the following network requirements 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli#meet-network-requirements' \n")
+                return consts.Diagnostic_Check_Failed
+
+        # For handling storage or OS exception that may occur during the execution
+        except OSError as e:
+                logger.warning("An exception has occured while performing the outbound connectivity check on the cluster. Exception: {}".format(str(e)) + "\n")
```

Comment:
Is this needed here since we're not doing any storing in local machine? I think we can remove checking for OSerror